### PR TITLE
Add java 10 Optional.orElseThrow()

### PR DESCRIPTION
**Pull request by treblereel**

### CODE REVIEW

**Comment by treblereel** [comment]
failed on:
```
external/com_google_j2cl/jre/java/super-wasm/exception_utils.js:18: ERROR - Property JSTag never defined on WebAssembly
const tag = WebAssembly.JSTag;
```

File: jre/javatests/com/google/j2cl/jre/java8/util/OptionalDoubleTest.java [hunk 1]
```diff
@@ -184,6 +184,18 @@ public void testOrElseThrow() {
...
+  public void testOrElseThrowNoArgs() {
+    try {
+      OptionalDouble.empty().orElseThrow();
+      fail("Expected NoSuchElementException from empty Optional: orElseThrow");
+    } catch (NoSuchElementException ignore) {
+      // expected
+    }
+
+    double value = OptionalDouble.of(10.0).orElseThrow();
+    assertEquals(10.0, value);
```
Comment (gkdn) [inline jre/javatests/com/google/j2cl/jre/java8/util/OptionalDoubleTest.java:196]: assertEquals(10.0, present.orElseThrow());

File: jre/javatests/com/google/j2cl/jre/java8/util/OptionalDoubleTest.java [hunk 2]
```diff
@@ -184,6 +184,18 @@ public void testOrElseThrow() {
     }));
   }
 
+  public void testOrElseThrowNoArgs() {
+    try {
+      OptionalDouble.empty().orElseThrow();
```
Comment (gkdn) [inline jre/javatests/com/google/j2cl/jre/java8/util/OptionalDoubleTest.java:189]: empty.orElseThrow()

**Comment by treblereel** [comment]
@gkdn thank you, fixed

**Pull request closed without merging**